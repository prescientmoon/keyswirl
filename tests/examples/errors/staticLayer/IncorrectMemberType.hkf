module exporting ()

layer foo:
| *_ => {}

template bar: a b c
layer bazz using bar: _ foo  _ -- foo is a layer, instead of a sequence
layer buzz using bar: _ (bazz "a")_ -- calling layers inside other layers is totally fine tho
