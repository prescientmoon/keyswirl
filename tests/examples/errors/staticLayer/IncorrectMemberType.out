[error WrongType]: Cannot match types
     +--> errors.staticLayer.IncorrectMemberType@7:25-7:28
     |
   7 | layer bazz using bar: _ foo  _ -- foo is a layer, instead of a sequence
     :       ^----             ^-- 
     :       |                 `- All elements inside a static layer must have type
     :       |                      Sequence
     :       |                    but this one has type
     :       |                      Chord -> Sequence
     :       |                    instead.
     :       `- while checking this static layer
-----+
